app/src/main/java/com/example/superimagesviewer/ui/InstagramPostFragment.kt

```diff
@@ -1,60 +1,66 @@
 package com.example.superimagesviewer.ui
 
 import android.os.Bundle
-import android.util.Log
 import androidx.fragment.app.Fragment
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
 import android.webkit.URLUtil
-import android.widget.Button
-import android.widget.EditText
+import androidx.databinding.DataBindingUtil
 import androidx.lifecycle.ViewModelProvider
-import androidx.navigation.findNavController
 import com.example.superimagesviewer.viewmodel.InstagramPostViewModel
 import com.example.superimagesviewer.R
+import com.example.superimagesviewer.databinding.InstagramPostFragmentBinding
+import kotlinx.coroutines.*
 
 class InstagramPostFragment : Fragment() {
 
-    private val TAG = InstagramPostFragment::class.simpleName
-
-    private lateinit var mosaicPageButton: Button
-    private lateinit var editTextPhotoUrl: EditText
-    private lateinit var uploadButton: Button
+    private lateinit var binding: InstagramPostFragmentBinding
     private lateinit var instagramPostViewModel: InstagramPostViewModel
 
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
-    ): View? {
-        return inflater.inflate(R.layout.instagram_post_fragment_, container, false)
+    ): View {
+        binding =
+            DataBindingUtil.inflate(inflater, R.layout.instagram_post_fragment, container, false)
+        return binding.root
     }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
-        editTextPhotoUrl = view.findViewById(R.id.edit_text_photo_url)
-        uploadButton = view.findViewById(R.id.upload_button)
-        mosaicPageButton = view.findViewById(R.id.post_to_mosaic_page_button)
-
-        mosaicPageButton.setOnClickListener {
-            view.findNavController().navigate(
-                InstagramPostFragmentDirections.actionInstagramPostFragmentToMosaicFragment()
-            )
-        }
-
-        uploadButton.setOnClickListener {
-            val photoUrl = editTextPhotoUrl.text.toString()
+        binding.uploadButton.setOnClickListener {
+            val photoUrl = binding.editTextPhotoUrl.text.toString()
             if (URLUtil.isValidUrl(photoUrl)) {
```

Comment:
Please swap if else statements like this:
if (!URLUtil.isValidUrl(photoUrl)) {
showDialog("Invalid or empty url")
return;
}
and here all your code if url is valid.

This required to simplify code readability and remove unnecessary braces

Reply:
Done